Textfield/Textfield.jsx

```diff
@@ -3,49 +3,120 @@ import MaterialComponent from "../MaterialComponent";
 import { MDCTextfield } from "@material/textfield/";
 
 /**
+ * @prop persistent = false
+ * @prop validation-msg = false
+ */
+class Helptext extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "textfield-helptext";
+    this._mdcProps = ["persistent", "validation-msg"];
+  }
+  materialDom(props) {
+    return (
+      <p {...props} aria-hidden="true">
+        {props.children}
+      </p>
+    );
+  }
+}
+
+/**
  * @prop fullwidth = false
  * @prop multiline = false
+ * @prop dense = false
+ * @prop disabled = false
  * @prop type = 'text'
  */
-export default class Textfield extends MaterialComponent {
+class TextfieldInput extends MaterialComponent {
   constructor() {
     super();
     this.componentName = "textfield";
-    this.state = {
-      showFloatingLabel: false
-    };
-    this._mdcProps = ["fullwidth", "multiline"];
+    this._mdcProps = ["fullwidth", "multiline", "dense", "disabled"];
   }
   componentDidMount() {
-    this.setState({
```

Comment:
this check was in place because floating labels(the way we are providing them) are only possible for JS enabled clients.

for a non jS client they should be outside the container

https://github.com/material-components/material-components-web/tree/master/packages/mdc-textfield#multi-line---gracefully-degraded